server/service/integration_core_test.go

```diff
@@ -7300,6 +7300,25 @@ func (s *integrationTestSuite) TestListSoftwareAndSoftwareDetails() {
 		"GET", fmt.Sprintf("/api/latest/fleet/software/versions/%d", versResp.Software[0].ID), nil, http.StatusNotFound, &detailsResp,
 		"team_id", "999999",
 	)
+
+	// a request with without_vulnerability_details set to false does not return extra details
+	respVersions := listSoftwareVersionsResponse{}
+	s.DoJSON(
+		"GET", "/api/latest/fleet/software/versions",
+		listSoftwareRequest{},
+		http.StatusOK, &respVersions,
```

Comment:
When using a premium flag/feature it appears the API returns an error and `fleet.ErrMissingLicense`. Wonder if that is something that should be done here as well.

Reply:
Silently dropping the option for Fleet Free, since theoretically you should be able to run Fleet Free without the EE code, feels fine here.

Also, in order to do the license error you'd need to be able to tell the difference between an unset query param and an explicit false, which would blow up complexity on this a bit, so just as well not to do that.